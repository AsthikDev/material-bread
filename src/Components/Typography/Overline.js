import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { StyleSheet } from 'react-native';
import BaseText from './BaseText';

const styles = StyleSheet.create({
  overline: {
    fontSize: 10,
    lineHeight: 10,
    letterSpacing: 1.5,
    textTransform: 'uppercase',
  },
});

export default class Overline extends Component {
  static propTypes = {
    children: PropTypes.node,
  };

  render() {
    return (
      <BaseText typographyStyles={styles.overline} {...this.props}>
        {this.props.children}
      </BaseText>
    );
  }
}
