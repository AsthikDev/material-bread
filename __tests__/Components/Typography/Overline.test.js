import React from 'react';
import Overline from '../../../src/Components/Typography/Overline';

import renderer from 'react-test-renderer';

test('it renders Overline', () => {
  const tree = renderer.create(<Overline>Overline</Overline>).toJSON();
  expect(tree).toMatchSnapshot();
});
