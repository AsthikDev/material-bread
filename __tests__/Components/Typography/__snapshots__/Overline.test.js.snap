// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders Overline 1`] = `
<Text
  style={
    Object {
      "color": "rgba(0, 0, 0, 0.87)",
      "fontSize": 10,
      "letterSpacing": 1.5,
      "lineHeight": 10,
      "textTransform": "uppercase",
    }
  }
  typographyStyles={
    Object {
      "fontSize": 10,
      "letterSpacing": 1.5,
      "lineHeight": 10,
      "textTransform": "uppercase",
    }
  }
>
  Overline
</Text>
`;
